### ISSUE TITLE
Using structs in functions which have lots of parameters

### ISSUE THREAD

**Issue by sahil21** [ElucidataInc/ElMaven#520]
There are a couple of functions where we are adding a lot of parameters. To make things more modular and readable, structs should be passed rather than passing lots of parameters/arguments. 
Some of the functions to start with:

- [ ] getEICs
- [ ] pullEICs
- [ ] groupPeaks

**Comment by chubukov** [comment]
For getEICs, if the idea is to turn the information about how the EIC was generated into a single object, then I would suggest using the `mzSlice` class, and extending it if necessary. After all, that's what it was intended to represent. Just an idea.

**Comment by chubukov** [comment]
For groupPeaks, it looks like most of the arguments could be stored in the mavenParameters class?

**Comment by sahil21** [comment]
@chubukov Agreed with getEIC. We should use `mzSlice`. It will make the code modular.

For `groupPeaks` and `pullEICs`, we can't use `mavenParameters` object. The reason being mavenParameters class will act as a controller now (MVC Architecture - we changed it while implementing dump & load settings).
